docs/source/links/state.md

```diff
@@ -0,0 +1,295 @@
+      };
+      cache.writeData({ data });
+      return data;
+    },
+  },
+});
+```
+
+To hook up your state link to Apollo Client, concatenate it to the other links
+in your Apollo Link chain. Your state link should be one of the last links in
+your chain. Then, pass your link chain to the Apollo Client constructor.
+
+```js
+const client = new ApolloClient({
+  cache: new InMemoryCache(),
+  link: stateLink.concat(new HttpLink()),
+});
+```
+
+How do we differentiate a request for local data from a request that hits our
+server? In our query or mutation, we specify which fields are client-only with a
+`@client` directive. This tells our network stack to retrieve or update the data
+in the cache with our resolver map that we passed into our state link.
+
+```js
+const UPDATE_NETWORK_STATUS = gql`
+  mutation updateNetworkStatus($isConnected: Boolean) {
+    networkStatus(isConnected: $isConnected) @client {
+      isConnected
+    }
+  }
+`;
+```
+
+To fire off the mutation from your component, bind your mutation to your
+component via your favorite Apollo view layer integration just like you normally
+would. Here's what this would look like for React:
+
+```js
+const WrappedComponent = graphql(UPDATE_NETWORK_STATUS, {
+  props: ({ mutate }) => ({
+    updateNetworkStatus: isConnected => mutate({ variables: { isConnected } }),
+  }),
+})(NetworkStatus);
+```
+
+Now that you've seen how easy it is to update local state with
+`apollo-link-state`, let's dive deeper into the resolver map.
+
+<h2 id="resolver">Resolver map</h2>
+
+Your resolver map is where all the magic happens to retrieve and update your
+local data in the Apollo cache. The resolver map is an object with resolver
+functions for each GraphQL object type. You can think of a GraphQL query or
+mutation as a tree of function calls for each field.
+
+Each field must either resolve to a GraphQL primitive (Int, String, etc) or
+resolve in a function call. You don't have to specify resolver functions for
+every field, however. GraphQL has [default
+resolvers](docs/graphql-tools/resolvers.html#Default-resolver). If the return
+value from the parent object has the same property names as the fields requested
+in the child object, you won't need to specify a resolver.
+
+```js
+const getUser = gql`
+  query {
+    user(id: 1) {
+      name {
+        last
+        first
+      }
+    }
+  }
+`;
+```
+
+For this query, you will need to specify a resolver for `Query.user` in your
+resolver map. If `Query.user` returns an object with a name property that
+corresponds to an object with last and first properties, you do not need to
+specify any additional resolvers. GraphQL takes care of that for you!
+
+The three most important things to keep in mind about resolvers in
+`apollo-link-state` are this:
+
+1. The cache is added to the context for you so you can write & read data from
+   the cache.
+2. The resolver should return an object with a `__typename` property. This is
```

Comment:
It's only necessary if your `dataIdFromObject` function uses `__typename`, so IDK

Reply:
Since the default `dataIdFromObject` uses `__typename`, I think this recommendation would apply to most people. I guess we can put a disclaimer like `The resolver should return an object with a __typename property unless your dataIdFromObject function doesn't use __typename for its cache keys.`?

Reply:
oof, that would be rough. Let's keep this for now...